.github/workflows/deploy.yml

```diff
@@ -67,84 +67,78 @@ jobs:
     needs: prep
     secrets: inherit
 
-  android:
-    name: Build and deploy Android
+  buildAndroid:
+    name: Build Android app
+    uses: ./.github/workflows/buildAndroid.yml
+    if: ${{ github.ref == 'refs/heads/staging' }}
     needs: prep
-    runs-on: ubuntu-latest-xl
+    secrets: inherit
+    with:
+      type: release
+      ref: staging
+
+  uploadAndroid:
+    name: Upload Android build to Google Play Store
+    needs: buildAndroid
+    runs-on: ubuntu-latest
     env:
       RUBYOPT: '-rostruct'
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Configure MapBox SDK
-        run: ./scripts/setup-mapbox-sdk.sh ${{ secrets.MAPBOX_SDK_DOWNLOAD_TOKEN }}
-
-      - name: Setup Node
-        uses: ./.github/actions/composite/setupNode
-
-      - name: Setup Java
-        uses: actions/setup-java@v4
-        with:
-          distribution: 'oracle'
-          java-version: '17'
-
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1.190.0
         with:
           bundler-cache: true
 
-      - name: Decrypt keystore
-        run: cd android/app && gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output my-upload-key.keystore my-upload-key.keystore.gpg
-        env:
-          LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
-
-      - name: Decrypt json key
-        run: cd android/app && gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output android-fastlane-json-key.json android-fastlane-json-key.json.gpg
-        env:
-          LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
-
-      - name: Get Android native version
-        id: getAndroidVersion
-        run: echo "VERSION_CODE=$(grep -o 'versionCode\s\+[0-9]\+' android/app/build.gradle | awk '{ print $2 }')" >> "$GITHUB_OUTPUT"
+      - name: Download Android build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: /tmp/artifacts
+          pattern: android-artifact-*
+          merge-multiple: true
 
-      - name: Build Android app
-        if: ${{ !fromJSON(env.SHOULD_DEPLOY_PRODUCTION) }}
-        run: bundle exec fastlane android build
-        env:
-          MYAPP_UPLOAD_STORE_PASSWORD: ${{ secrets.MYAPP_UPLOAD_STORE_PASSWORD }}
-          MYAPP_UPLOAD_KEY_PASSWORD: ${{ secrets.MYAPP_UPLOAD_KEY_PASSWORD }}
+      - name: Log downloaded artifact paths
+        run: ls -R /tmp/artifacts
```

Comment:
Do we still need this? Maybe let's make a follow up to remove.

Reply:
I can remove it if you prefer, but I was also thinking why not just let it stay? It's not really doing any harm there, kind of makes the directory structure clear and shows you the name of files that are currently only represented as variables in the code 🙂